docs/en/operations/backup.md

```diff
@@ -19,23 +19,193 @@ Keep in mind that if you backed something up and never tried to restore it, chan
+```sql
+BACKUP DATABASE default, TABLE system.users, \
+  TABLE system.roles, TABLE system.settings_profiles, \
+  TABLE system.row_policies, TABLE system.quotas \
+  TO Disk('backups', '1.zip') ASYNC
+```
+
+Backup the `test` database with a new name (`test2`):
+```sql
+BACKUP DATABASE test AS test2 TO Disk('backups', '1.zip')
+```
+
+Backup the database `test`:
+```sql
+BACKUP DATABASE test TO Disk('backups', '1.zip')
+
+## BACKUP TABLE
+
+Backup the table `mv_1`:
+```sql
+BACKUP TABLE mv_1 TO Disk('backups', '1.zip')
+```
+
+Backup several of the `system` tables:
+```sql
+BACKUP TABLE system.users, TABLE system.roles, \
+  TABLE system.settings_profiles, TABLE system.row_policies, \
+  TABLE system.quotas TO Disk('backups', '1.zip')
+```
+
+Backup with a specific username:
+```sql
+BACKUP TABLE system.users, TABLE system.roles \
+  TO Disk('backups', '1.zip'), user="u2"
+```
+
+Backup a table with a new name:
+```sql
+BACKUP TABLE test.table AS test.table2 TO Disk('backups', '1.zip')
+```
+
+Backup a table asynchronously:
+```sql
+BACKUP TABLE test.table TO Disk('backups', '1.zip') ASYNC
+```
+
+### Compress and password protect a backup
+
+Use `lzma` compression, and password protect the backup:
+```sql
+BACKUP TABLE test.table TO Disk('backups', '1.zip') \
+  SETTINGS compression_method='lzma', compression_level=3, password='qwerty'
+```
+
+Add an `id` to the backup (the id is used during `RESTORE`):
+```sql
+BACKUP TABLE test.table TO Disk('backups', '1.zip') \
+  SETTINGS id='first' ASYNC
+```
+
+### Incremental table backup
+
+Incrementally backup a table:
+```sql
+BACKUP TABLE test.table2 TO {incremental_backup_name} \
+  SETTINGS base_backup = Disk('backups', '1.zip')
+```
+
+## Restore
+
+Restore specific partitions:
+```sql
+RESTORE TABLE test.table PARTITIONS '2', '3' FROM Disk('backups', '1.zip')
+```
+
+Restore all databases and tables from a specific backup:
+```sql
+RESTORE ALL FROM Disk('backups', '1.zip')
+```
+
+Restore based on a username:
+```sql
+RESTORE ALL FROM Disk('backups', '1.zip'), user="u1"
+
+Restore specific databases and tables asynchronously:
+```sql
+RESTORE DATABASE default, TABLE system.users, TABLE system.roles, TABLE system.settings_profiles, TABLE system.row_policies, TABLE system.quotas FROM Disk('backups', '1.zip') ASYNC
+```
+
+Restore and rename a table:
+```sql
+RESTORE DATABASE test2 AS test3 FROM Disk('backups', '1.zip')
+```
+
+Restore specific tables:
+```sql
+RESTORE TABLE system.users, TABLE system.roles, TABLE system.settings_profiles, TABLE system.row_policies, TABLE system.quotas FROM Disk('backups', '1.zip')
+```
+
+```sql
+RESTORE TABLE test.table2 FROM {incremental_backup_name}
+```
+
+Restore and rename while specifying a username:
+```sql
+RESTORE TABLE test.table AS test.table2 FROM Disk('backups', '1.zip'), user="u1"
+```
+
+Restore and rename from an incremental backup:
+```sql
+RESTORE TABLE test.table AS test.table2 FROM {incremental_backup_name}"
```

Comment:
As far as I understand this is no different from restoring from a full backup

Reply:
Thinking back to restoring from incremental backups on tape (no laughing), The incrementals get restored in order by date ascending.  What we need to know is: does the backup system keep a list of all of the incrementals and then does the restore command get that list and apply in order?

Reply:
An incremental backup contains a reference to its base (which is another backup). It has nothing to do with the actual order of restoring, it's just a way to make backups smaller because an incremental backup doesn't need to contain all its data, it contains only new or changed files instead. An incremental backup keeps a reference to its base backup inside so it's not necessary to write `SETTINGS base_backup=` when we execute the RESTORE command.